### PR TITLE
Refactor save interval and eval interval to share code

### PR DESCRIPTION
**Pull request by dakinggg**
This PR refactors the interval scheduler helper functions used for checkpointing and eval to use the same function. We can now use this unified function anywhere we need an interval scheduler, and only need to add new features in one place. An immediate outcome of this is that we can use `dur` interval scheduling for checkpointing now.

### CODE REVIEW

File: composer/core/evaluator.py [hunk 1]
```diff
@@ -195,10 +107,13 @@ def eval_interval(self):
 
     @eval_interval.setter
     def eval_interval(self, eval_interval: Optional[Union[int, str, Time, Callable[[State, Event], bool]]]):
+        from composer.utils import create_interval_scheduler
```
Comment (mvpatel2000) [inline composer/core/evaluator.py:110]: This does not need to be inlined since core depends on utils

File: tests/trainer/test_trainer_eval.py [hunk 2]
```diff
@@ -462,10 +462,10 @@ def test_eval_at_fit_end(eval_interval: Union[str, Time, int], max_duration: str
...
     )
 
-    evaluator.eval_interval = evaluate_periodically(
-        eval_interval=eval_interval,
-        eval_at_fit_end=eval_at_fit_end,
-    )
+    evaluator.eval_interval = create_interval_scheduler(interval=eval_interval,
+                                                        include_end_of_training=eval_at_fit_end,
+                                                        checkpoint_events=False,
+                                                        final_events={Event.FIT_END})
```
Comment (mvpatel2000) [inline tests/trainer/test_trainer_eval.py:468]: can we restore trailing comma so it's cleaner?

**Review by mvpatel2000 — COMMENTED** [review]
minor nits. mostly lgtm

File: composer/callbacks/utils.py [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
-from composer.core import Event, State, Time, TimeUnit
+from composer.core import Event, State, Time
+from composer.utils.misc import create_interval_scheduler as _create_interval_scheduler
 
 
 def create_interval_scheduler(interval: Union[str, int, Time],
-                              include_end_of_training=True) -> Callable[[State, Event], bool]:
+                              include_end_of_training: bool = True,
+                              checkpoint_events: bool = True,
+                              final_events: Optional[Set[Event]] = None) -> Callable[[State, Event], bool]:
```
[the remark this answers is not in the document]
Comment (dakinggg) [reply composer/callbacks/utils.py:16]: but i need to get a deprecation warning if anyone actually tries to use it, right?

**Review by mvpatel2000 — APPROVED** [review]
Very nice 🎉

**Pull request merged**